### PR TITLE
fix(guide) Replace invalid prism code block names

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [ ] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Related to #35960

This PR aims to resolve 95% of the issues related to issue #35960.  I will create a separate PR for the other 5% based on feedback given to my comment on the issue.

Since my changes only effect code blocks, I put everything on a single PR (note I did not check the box above regarding multiple languages on the same PR).  If these need to be broken down by language, I can do that if absolutely necessary.

I have kept all the commits (one for each type of replacement made) in case I need to revert any.  At the very end, I can squash them if necessary.

### CODE REVIEW

**Comment by RandellDawson** [comment]
OK, I no longer see the prism warnings after my latest commits.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Nice work!  There were a few \```sh and ```C++ that slipped through, but for the most part it looks great.  I didn't check the formatting of every file, but I did look at a semi-random sample and they all seemed fine.  After our discussion, I checked the XAML -> xml change and that looks fine to me.

One thing I wasn't sure how to do was build the guide in any language other than English.  Any ideas?

Just restate, because the comment order isn't what I thought it would be: there were some `jsx`s used in the redux section, but that's just pure JS.

File: guide/spanish/rust/installing-rust/index.md [hunk 1]
```diff
@@ -22,7 +23,7 @@ Esto buscará el instalador de `rustup` y, a su vez, `rustup` todo lo que necesi
 # Verificando instalación
 
 La instalación de `rustup` instalará todas las cosas relevantes para la oxidación, pero lo más importante es que se instale el compilador y el administrador de paquetes. Para verificar que todo está instalado, ejecute este comando:
-
+```shell
 ```sh
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/spanish/rust/installing-rust/index.md]: @ojeytonwilliams According to prism's supported languages, `sh` is not one of them.  We can use `bash` or `shell` for Bash.

File: guide/chinese/certifications/front-end-libraries/redux/dispatch-an-action-event/index.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ localeTitle: 派遣行动事件
 
 通过调用dispatch方法将LOGIN操作发送到Redux存储，并传入`loginAction()`创建的操作。
 
-```react.js
+```jsx
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/chinese/certifications/front-end-libraries/redux/dispatch-an-action-event/index.md]: @ojeytonwilliams `jsx` is supported by prism, so why not use it?

File: guide/chinese/certifications/front-end-libraries/redux/get-state-from-the-redux-store/index.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ localeTitle: 从Redux商店获取状态
 
 使用`getState()`方法从存储中检索数据。
 
-```react.js
+```jsx
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/chinese/certifications/front-end-libraries/redux/get-state-from-the-redux-store/index.md]: @ojeytonwilliams `jsx` is supported by prism, so why not use it?

File: guide/russian/cplusplus/terms-to-know-for-beginners/index.md [hunk 4]
```diff
@@ -33,7 +34,7 @@ _В: Попробуйте найти IDE в Google и запустить на н
 ```
 
 Вышеприведенный код возвращает ошибку, потому что в строке 2 мы использовали двоеточие (:) вместо точки с запятой (;) Итак, давайте отлаживаем ошибку:
-
+```cpp
 ```C++
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/russian/cplusplus/terms-to-know-for-beginners/index.md]: Prism documentation shows `C++` needs to be `cpp`

File: guide/spanish/cplusplus/terms-to-know-for-beginners/index.md [hunk 5]
```diff
@@ -33,7 +34,7 @@ Un programa de muestra:
 ```
 
 El código anterior devuelve un error porque en la línea 2, hemos usado dos puntos (:) en lugar de un punto y coma (;) Entonces, vamos a depurar el error:
-
+```cpp
 ```C++
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/spanish/cplusplus/terms-to-know-for-beginners/index.md]: Prism documentation shows C++ needs to be cpp

File: guide/spanish/csharp/hello-world/index.md [hunk 6]
```diff
@@ -28,7 +29,7 @@ using System;
 ```
 
 ## Salida:
-
+```shell
 ```sh
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/spanish/csharp/hello-world/index.md]: see my other related comment

File: guide/spanish/rust/installing-rust/index.md [hunk 7]
```diff
@@ -12,7 +13,7 @@ Visite el [sitio web de rustup](https://rustup.rs) y descargue el `rustup-init.e
 ## Instalación de Rust en otros sistemas operativos (Mac OS X, Linux, BSD, Unix)
 
 Abre tu terminal y escribe este comando:
-
+```shell
 ```sh
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/spanish/rust/installing-rust/index.md]: see my other related comment

File: guide/spanish/redux/tutorial/index.md [hunk 8]
```diff
@@ -12,7 +13,7 @@ Se basa en el principio de que una aplicación [Node.js](https://nodejs.org/) ya
 Suponiendo que todo está configurado y funcionando correctamente, hay algunos paquetes que deben agregarse para que Redux funcione con React.
 
 Abra un terminal dentro de la carpeta del proyecto que se creó y ejecute el siguiente comando
-
+```shell
 ```sh
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/spanish/redux/tutorial/index.md]: see my other related comment

File: guide/chinese/rust/installing-rust/index.md [hunk 9]
```diff
@@ -12,7 +13,7 @@ localeTitle: 安装Rust
 ## 在其他操作系统中安装Rust（Mac OS X，Linux，BSD，Unix）
 
 打开终端并输入以下命令：
-
+```shell
 ```sh
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/chinese/rust/installing-rust/index.md]: see my other related comment

File: guide/chinese/cplusplus/terms-to-know-for-beginners/index.md [hunk 10]
```diff
@@ -33,7 +34,7 @@ _问：尝试在Google上搜索IDE并在其上运行您的第一个程序。检
 ```
 
 上面的代码返回一个错误，因为在第2行，我们使用了冒号（:)而不是分号（;） 那么，让我们调试错误：
-
+```cpp
 ```C++
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/chinese/cplusplus/terms-to-know-for-beginners/index.md]: see my other related comment

File: guide/chinese/certifications/front-end-libraries/redux/get-state-from-the-redux-store/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply guide/chinese/certifications/front-end-libraries/redux/get-state-from-the-redux-store/index.md]: See above

File: guide/chinese/certifications/front-end-libraries/redux/dispatch-an-action-event/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (RandellDawson) [reply guide/chinese/certifications/front-end-libraries/redux/dispatch-an-action-event/index.md]: Fair enough.

**Comment by RandellDawson** [comment]
@ojeytonwilliams  Thanks for the review.  Not sure how those extra lines got in there, but I must have messed up my regex replace in VS Code somehow.  I have committed all of your suggestions.

> One thing I wasn't sure how to do was build the guide in any language other than English. Any ideas?

I don't think the logic has been completely implemented yet but @raisedadead could answer better.

**Comment by raisedadead** [comment]
You can locally build the guide or learn for available languages by changing the `LOCALE=english` in your `.env` files.

**Comment by RandellDawson** [comment]
@raisedadead followup question:  When we use `npm run develop`, then guide is really just the `mock-guide` files.  Is there a way use `npm run develop` so that the complete guide is built out for a particular language using the actual guide folder?

**Comment by ojeytonwilliams** [comment]
@RandellDawson You could edit `gatsby-config.js` as a quick hack.  There's a line 
```js
const selectedGuideDir = `../${
  env === 'production' ? 'guide' : 'mock-guide'
}/${locale}`;
```
That you'd want to change.

@raisedadead Yeah, I tried that, but there are errors in the challenge files, so `gatsby develop` never finishes.  I guess the solution is to just not build the challenges.  I'll see if I can do that without breaking things!

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by raisedadead** [comment]
With languages other than English you have to build the production bundle. Spoiler you need a really good machine that will survive the build. 

Then you can spawn the api app and the client app separately.

You want `gatsby serve` in client and dig into the server package json for the script to spin up the api.

**Comment by moT01** [comment]
You can copy all the stuff from the guide folder and paste in the mock-guide folder - that's how I usually test guide changes.

**Comment by RandellDawson** [comment]
**Moderators:**  It appears we are going to have to wait a while to merge this one, because there are several PRs (across multiple languages before it which would result in merge conflicts) if merged first. I will start reviewing the English guide articles, but we will need members of the non-English language teams to help review the other non-English PRs.

**Comment by moT01** [comment]
looking at the contributors tool @RandellDawson - it looks like maybe 50 files have other PR's touching these files - you could remove those files from this PR and there would be no potential for conflicts - and we could maybe merge this faster

**Comment by RandellDawson** [comment]
@moT01 I suppose I can revert the changes to the files on those specific earlier PRs and then go into each and add the prism support language to the triple backticks if needed.  That way, they will get the update, but not end up with a merge conflict.  What do you think?

**Comment by RandellDawson** [comment]
I have successfully reverted the changes to the 42 files on earlier PRs which would have caused merge conflicts if this PR would have been merged.  Now, I will review the 68 PRs which had these 42 files and add an extra commit to fix the invalid prism code block names if needed.

**Comment by RandellDawson** [comment]
OK.  I just went through the 68 PRs which contained the files 42 files I reverted earlier.  I made commits on those PRs so if they are merged, they will already have the correct code block names.

**Review by moT01 — CHANGES_REQUESTED** [review]
I took a look through these - nice job @RandellDawson! I found these few things that I think got missed. Other than those I think the rest looks good.

I did not test this locally - I simply looked through the diff

**Comment by RandellDawson** [comment]
@moT01 OK, I committed the changes which I had missed.  I am thinking we should probably have one more moderator take a look at this PR before it gets merged.

**Review by moT01 — APPROVED** [review]

**Review by Manish-Giri — APPROVED** [review]
LGTM 💯

**Comment by ojeytonwilliams** [comment]
@RandellDawson you mentioned taking another look - are you happy with the PR?

**Comment by RandellDawson** [comment]
@ojeytonwilliams Yes I am happy with it now.  Feel free to merge it. Once this one is merged, I will be creating another PR for a related issue which deals with a ton of files which have a space between the triple backticks and the language specifier (like below).
````
``` js
// code goes here
```
````

**Comment by ojeytonwilliams** [comment]
Thanks @RandellDawson for the hard work and @moT01 and @Manish-Giri for reviewing this monster!

**Pull request merged**